### PR TITLE
PP-5391 add reference to refund event payload

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- add reference to refund events payload
- update relevant tests

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/events/model/refund/RefundEvent.java [hunk 1]
```diff
@@ -9,15 +9,19 @@
 public abstract class RefundEvent extends Event {
 
     private String parentResourceExternalId;
+    private String reference;
```
Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/events/model/refund/RefundEvent.java]: reference should be in the event details, not in the event itself

File: src/test/java/uk/gov/pay/connector/events/model/refund/RefundEventFactoryTest.java [hunk 2]
```diff
@@ -42,13 +44,14 @@ public void givenARefundCreatedTypeShouldCreateARefundCreatedByUserEvent() {
     public void givenANonPayloadPaymentEventTypeShouldCreateTheCorrectPaymentEventType() {
         RefundHistory refundHistory = new RefundHistory(1L, "external_id", 50L, RefundStatus.REFUND_SUBMITTED.getValue(),
                 charge.getId(), new UTCDateTimeConverter().convertToDatabaseColumn(createdDate), 1L,
-                "reference", new UTCDateTimeConverter().convertToDatabaseColumn(createdDate.plusSeconds(1L)),
+                "refund_reference", new UTCDateTimeConverter().convertToDatabaseColumn(createdDate.plusSeconds(1L)),
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/events/model/refund/RefundEventFactoryTest.java:45]: I think title of method could change as it is no longer NonPayload event ?
possibly `givenARefundSubmittedEventTypeShouldCreateTheCorrectPaymentEventType` or we can create parameterised test for all refund events (refund submitted, error, succeeded) that includes `reference` in the payload for test completeness

**Review by kbottla — COMMENTED** [review]
LGTM! Just added a minor comment about test naming and using parameterised test to test all refund events that includes reference in the payload

**Review by kbottla — APPROVED** [review]
LGTM!

**Pull request merged**